phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexUpgradeToolIT.java

```diff
@@ -0,0 +1,506 @@
+                }
+            } else {
+                args.add("rollback");
+                if (mutable) {
+                    for (String table : tableList) {
+                        Assert.assertTrue(admin.getDescriptor(TableName.valueOf(table)).hasCoprocessor(IndexRegionObserver.class.getName()));
+                        Assert.assertFalse(admin.getDescriptor(TableName.valueOf(table)).hasCoprocessor(Indexer.class.getName()));
+                    }
+                }
+                for (String index : indexesList) {
+                    Assert.assertTrue(admin.getDescriptor(TableName.valueOf(index)).hasCoprocessor(GlobalIndexChecker.class.getName()));
+                }
+            }
+            BufferedWriter writer = new BufferedWriter(new FileWriter(new File("/tmp/input_file_index_upgrade.csv")));
+            writer.write(inputList);
+            writer.close();
+            args.add("-f");
+            args.add("/tmp/input_file_index_upgrade.csv");
+            args.add("-c");
+            args.add(conn.unwrap(PhoenixConnection.class).getURL());
+            IndexUpgradeTool.main(args.toArray(new String[args.size()]));
+            if (upgrade) {
+                if (mutable) {
+                    for (String table : tableList) {
+                        Assert.assertTrue(admin.getDescriptor(TableName.valueOf(table)).hasCoprocessor(IndexRegionObserver.class.getName()));
+                        Assert.assertFalse(admin.getDescriptor(TableName.valueOf(table)).hasCoprocessor(Indexer.class.getName()));
+                    }
+                }
+                for (String index : indexesList) {
+                    Assert.assertTrue(admin.getDescriptor(TableName.valueOf(index)).hasCoprocessor(GlobalIndexChecker.class.getName()));
+                }
+            } else {
+                if (mutable) {
+                    for (String table : tableList) {
+                        Assert.assertTrue(admin.getDescriptor(TableName.valueOf(table)).hasCoprocessor(Indexer.class.getName()));
+                        Assert.assertFalse(admin.getDescriptor(TableName.valueOf(table)).hasCoprocessor(IndexRegionObserver.class.getName()));
+                    }
+                }
+                for (String index : indexesList) {
+                    Assert.assertFalse(admin.getDescriptor(TableName.valueOf(index)).hasCoprocessor(GlobalIndexChecker.class.getName()));
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testToolWithShowParameter() throws IOException, SQLException {
+        try(Connection conn = DriverManager.getConnection(getUrl(), new Properties())) {
+            ConnectionQueryServices queryServices = conn.unwrap(PhoenixConnection.class).getQueryServices();
+            Admin admin = queryServices.getAdmin();
+
+            ArrayList<String> args = new ArrayList<>();
+            args.add("-o");
+            if (upgrade) {
+                args.add("upgrade");
```

Comment:
Declare a const for "upgrade" / "rollback" and reuse between tool and test-code. Perhaps do the same for all literals used multiple times.